moped-editor/src/views/projects/projectView/ProjectPhases.js

```diff
@@ -1,411 +1,232 @@
-                project_phase_id,
-                data.moped_proj_phases
-              );
-              if (projPhasesIdsToUpdate.length > 0) {
-                await updateExistingPhases(projPhasesIdsToUpdate);
-              }
-            }
-
-            // Execute update mutation
-            await updateProjectPhase({
-              variables: { project_phase_id, object: updatedPhasePayload },
-            }).catch((err) => {
-              console.error(err);
-            });
-            // Refetch data
-            await refetch();
-
-            if (!!projectViewRefetch) {
-              await projectViewRefetch();
-            }
-            setIsMutating(false);
-          },
-          onRowDelete: async (oldData) => {
-            // Execute delete mutation
-            setIsMutating(true);
-            await deleteProjectPhase({
-              variables: {
-                project_phase_id: oldData.project_phase_id,
-              },
-            }).catch((err) => {
-              console.error(err);
-            });
-            await refetch();
-
-            if (!!projectViewRefetch) {
-              await projectViewRefetch();
-            }
-            setIsMutating(false);
-          },
+      <DataGrid
+        autoHeight
+        columns={columns}
+        density="comfortable"
+        getRowId={(row) => row.project_phase_id}
+        disableRowSelectionOnClick
+        getRowHeight={() => "auto"}
+        hideFooterPagination={true}
+        localeText={{ noRowsLabel: "No phases" }}
+        rows={data?.moped_proj_phases || []}
+        slots={{
+          toolbar: ProjectPhaseToolbar,
         }}
-        title={
-          <Typography variant="h2" color="primary">
-            Phases
-          </Typography>
-        }
-        options={{
-          ...(data.moped_proj_phases.length < PAGING_DEFAULT_COUNT + 1 && {
-            paging: false,
-          }),
-          search: false,
-          rowStyle: { fontFamily: typography.fontFamily },
-          actionsColumnIndex: -1,
-          addRowPosition: "first",
-          idSynonym: "project_phase_id",
-        }}
-        localization={{
-          header: {
-            actions: "",
-          },
-          body: {
-            emptyDataSourceMessage: (
-              <Typography variant="body1">
-                No project phases to display
-              </Typography>
-            ),
+        slotProps={{
+          toolbar: {
+            addAction: onClickAddPhase,
+            setIsDialogOpen: setIsTemplateDialogOpen,
           },
         }}
       />
+      {editPhase && (
+        <ProjectPhaseDialog
+          phase={editPhase}
+          onClose={() => setEditPhase(null)}
+          onSubmitCallback={onSubmitCallback}
+          phases={data?.moped_phases}
+          currentProjectPhaseIds={currentProjectPhaseIds}
+          projectId={projectId}
+        />
+      )}
       <PhaseTemplateModal
```

Comment:
**Requested change** When I tested adding from template, I'm seeing an error related to the `current_phase_ids_to_clear`. Probably a small adjustment needed to the state or mutation that fires.
<img width="455" alt="Screenshot 2024-01-16 at 4 28 30 PM" src="https://github.com/cityofaustin/atd-moped/assets/37249039/7af16952-0be6-4822-9f54-c4d0efb4b5ef">